fixed header for mobile devices

chaitak-gorai opened the pull request:
## Description
- The header was not responsive for mobile devices. Solved this using tailwind.

## Solution:
- Added a Hamburger Button for mobile devices which when clicked expanded the nav-items menu.

## FIles Changed
- Header.vue (CSS only)

## Issues Solved
- Resolves #32 

## Preview (The issue)
![mobile_header](https://user-images.githubusercontent.com/77141674/229299318-b03887c9-986b-4081-bb10-2078db2f411f.png)

## Preview After Solving
![after1](https://user-images.githubusercontent.com/77141674/229299394-5076380b-0122-4e83-9a41-54ef1811b746.png)
![after2](https://user-images.githubusercontent.com/77141674/229299409-d954151c-ffb7-4ff5-8767-093b7957fe25.png)
![after3](https://user-images.githubusercontent.com/77141674/229299412-f673ec00-0064-4816-b019-6cb71a473db4.png)

chaitak-gorai committed:
fixed head for mobile devices

muarachmann commented on resources/app/shared/includes/Header.vue:
```diff
@@ -1,380 +1,830 @@
+                        <svg
+                            class="h-5 w-5"
+                            stroke="currentColor"
+                            xmlns="http://www.w3.org/2000/svg"
```
Please use the `icon` component here for adding icons to the repo. We use https://heroicons.com for our reference.

chaitak-gorai replied:
Sure, I am doing the required changes,

muarachmann commented on resources/app/shared/includes/Header.vue:
```diff
@@ -1,380 +1,830 @@
+                                    class="h-6 w-6 fill-current"
+                                    viewBox="0 0 24 24"
+                                    xmlns="http://www.w3.org/2000/svg"
+                                >
```
Same here

chaitak-gorai committed:
Added icon template for svg icons

chaitak-gorai commented:
## Update
- Hello @muarachmann , as requested, i have used the `icon` component for the SVG icons. Some icons were missing in the `Icon.vue` , so added them to the template

## FIles Changed
- Header.vue 
- Icon.vue

### Please review the code and suggest to me if some further changes are required.

muarachmann commented:
This looks good just a few alignment with the `x` (close) button. @chaitak-gorai are you considering applying for GSoC?

chaitak-gorai commented:
> This looks good just a few alignment with the `x` (close) button. @chaitak-gorai are you considering applying for GSoC?

Yes, I am applying for GSoC, and I find this project incredibly interesting. It's overwhelming to think about how my code could contribute to humanity through the health sector.

muarachmann commented:
Thanks @chaitak-gorai while this menu is good. I will rather prefer a drawer menu with menu items. and submenus You can consider max level 2 for the sub-menus

chaitak-gorai commented:
> Thanks @chaitak-gorai while this menu is good. I will rather prefer a drawer menu with menu items. and submenus You can consider max level 2 for the sub-menus

Ok.. I will modify the menu, and will use a drawer menu.

muarachmann commented:
Any updates on this? @chaitak-gorai

chaitak-gorai commented:
> Any updates on this? @chaitak-gorai

Hello @muarachmann, sorry for the delay, I was not available due to a family emergency. Will work on it now. I will show you the updated design by tomorrow.

chaitak-gorai committed:
✨ Drawer Navigation Menu

chaitak-gorai commented:
## Update
- Hello @muarachmann , as requested, i have updated the menu into a drawer menu. The drawer will slide from left to right when clicking the hamburger button. It has 2 levels for menus and submenus.

## FIles Changed
- tailwind.config: Added a tailwind component package `flowbite`
- Header.vue : Updated the UI
- Icon.vue : Added some new Icons

## Screenshot
![updated](https://user-images.githubusercontent.com/77141674/232918944-0cf40f01-a6ff-4575-b8c5-f9893ca4417a.png)

### Please look over the code and let me know if any more changes are required.

muarachmann commented:
chevron should be normal without bg. Like on the homescreen. Aside that is there any reason why we had to pull in flowbite?
Also tapping out of the menu doesn't close the menu

chaitak-gorai committed:
Fixed chevron icon color

chaitak-gorai commented:
## Update

@muarachmann , I have made the requested icon update, and now the menu can be closed by tapping outside the screen.

We are utilizing flowbite, a component library built over tailwindcss, to implement interactive sections such as the drawer menu and submenus with improved design and style. This will not only enhance the UI but also accelerate the process of creating other UI sections during the migration from legacy code to the Laravel framework.

Additionally, please let me know if this update meets your requirements or if any further changes are needed.

## Screenshot
![image](https://user-images.githubusercontent.com/77141674/233565777-2e955be1-7228-4572-b3d5-52cc1c7f4227.png)

muarachmann commented:
Nice @chaitak-gorai one last thing is the drawer when closed prevents interaction. Looks like the overlay is still there.

chaitak-gorai commented:
Hey @muarachmann sorry but i couldn't find the issue on my end. 
I am attaching the screen. 
Can you please elaborate what is the possible issue here?.

https://user-images.githubusercontent.com/77141674/233722005-7b8530d6-5aca-4f59-ab6a-9e819cbec69b.mp4

 capture.

muarachmann commented:
@chaitak-gorai tap on the body not the x to dismiss the modal, it dismisses but overlay is still there and prevents interaction

chaitak-gorai commented:
> @chaitak-gorai tap on the body not the x to dismiss the modal, it dismisses but overlay is still there and prevents interaction

yeah got it thanks. :)
fixing it.

chaitak-gorai committed:
🐛 fixed the backdrop bug

chaitak-gorai commented:
## Fixed 

I have fixed the bug of the backdrop. Now the overlay doesn't exist after the drawer is closed by clicking outside it.

Please review it and suggest if any further change is required.

https://user-images.githubusercontent.com/77141674/233744579-169c668b-5af2-4e6c-b2cc-3f494ba6e350.mp4

muarachmann commented:
LGMT
